cmd/traefik/traefik.go

```diff
@@ -45,19 +45,17 @@ Complete documentation is available at https://traefik.io`,
 		Config:                traefikConfiguration,
 		DefaultPointersConfig: traefikPointersConfiguration,
 		Run: func() error {
-			globalConfiguration := traefikConfiguration.GlobalConfiguration
-			if globalConfiguration.File != nil && len(globalConfiguration.File.Filename) == 0 {
-				// no filename, setting to global config file
-				if len(traefikConfiguration.ConfigFile) != 0 {
-					globalConfiguration.File.Filename = traefikConfiguration.ConfigFile
-				} else {
-					log.Errorln("Error using file configuration backend, no filename defined")
-				}
-			}
-			if len(traefikConfiguration.ConfigFile) != 0 {
-				log.Infof("Using TOML configuration file %s", traefikConfiguration.ConfigFile)
+			globalConfiguration := &traefikConfiguration.GlobalConfiguration
+			fileProviderEnabled := globalConfiguration.File != nil
+			traefikConfigFileExists := len(traefikConfiguration.ConfigFile) != 0
+
+			// Try to fallback to traefik config file in case the file provider is enabled
+			// but it has no file name configured.
+			if fileProviderEnabled && len(globalConfiguration.File.Filename) == 0 && traefikConfigFileExists {
+				globalConfiguration.File.Filename = traefikConfiguration.ConfigFile
```

Comment:
Can you move this code in the run func, and use the "new" config file param

Reply:
I went one step further and pushed it down to `SetEffectiveConfiguration` as it fits there best IMO.